### PR TITLE
Support group arg for conv grad in CPU backend

### PR DESCRIPTION
**Pull request by beicy**

### CODE REVIEW

File: tests/unittests/gradCheckTest.cpp [hunk 1]
```diff
@@ -199,11 +200,29 @@ static void gradientCheckGroupConv(size_t numInputChan, size_t group) {
...
+TEST(Network, gradientCheckDepthwiseConv) {
+  gradientCheckGroupConv(4, 4, BackendKind::Interpreter);
+}
+
+TEST(Network, gradientCheckGroupConv) {
+  gradientCheckGroupConv(4, 2, BackendKind::Interpreter);
+}
 
-TEST(Network, gradientCheckDepthwiseConv) { gradientCheckGroupConv(4, 4); }
+TEST(Network, gradientCheckConvCPU) {
```
Comment (bertmaher) [inline tests/unittests/gradCheckTest.cpp]: Consider using a parameterized test fixture (`TEST_P`) for the different backends, like we use in OperatorTest and MLTest.  It can be a separate PR if you'd like.

File: tests/unittests/gradCheckTest.cpp [hunk 2]
```diff
@@ -199,11 +200,29 @@ static void gradientCheckGroupConv(size_t numInputChan, size_t group) {
   performGradCheck(IP, result, A, Ex, &inputs, &outputs, 0.001, 0.004);
 }
 
-TEST(Network, gradientCheckConv) { gradientCheckGroupConv(1, 1); }
+TEST(Network, gradientCheckConv) {
```
Comment (rdzhabarov) [inline tests/unittests/gradCheckTest.cpp]: can you use a similar technique as in operatorTest when we support multiple backends for tests?

File: tests/unittests/gradCheckTest.cpp (hunk 1 above)
Comment (rdzhabarov) [reply tests/unittests/gradCheckTest.cpp]: that was fast :)

**Comment by rdzhabarov** [comment]
There is an exception while running under GCC: https://travis-ci.org/pytorch/glow/jobs/400583898

File: tests/unittests/gradCheckTest.cpp [hunk 3]
```diff
@@ -26,6 +26,14 @@
 
 using namespace glow;
 
+class GradCheckBase : public ::testing::TestWithParam<BackendKind> {
+public:
+  ExecutionEngine IP_{GetParam()};
```
Comment (rdzhabarov) [inline tests/unittests/gradCheckTest.cpp]: nit: IP seems to be coming from "interpreter", maybe call this backend_.

Comment (beicy) [reply tests/unittests/gradCheckTest.cpp]: "IP" stands for Execution Engine in the code.  Will change it to EE_.

File: lib/Backends/CPU/libjit/libjit_conv.cpp [hunk 4]
```diff
@@ -492,49 +492,58 @@ void libjit_convolution_grad_f(float *inG, const float *outG, const float *inW,
...
+                }
+
+                for (size_t c = 0; c < inCperG; c++) {
+                  inG[libjit_getXYZW(inWdims, n, (size_t)ax, (size_t)ay,
+                                     g * inCperG + c)] +=
+                      filterW[libjit_getXYZW(filterGdims, d, kx, ky, c)] * grad;
+                  filterG[libjit_getXYZW(filterGdims, d, kx, ky, c)] +=
+                      inW[libjit_getXYZW(inWdims, n, (size_t)ax, (size_t)ay,
+                                         g * inCperG + c)] *
+                      grad;
```
Comment (rdzhabarov) [inline lib/Backends/CPU/libjit/libjit_conv.cpp]: nit: fix format.

File: tests/unittests/gradCheckTest.cpp [hunk 5]
```diff
@@ -196,22 +201,22 @@ static void gradientCheckGroupConv(size_t numInputChan, size_t group) {
   inputsH.initXavier(1, mod.getPRNG());
   outputsH.initXavier(1, mod.getPRNG());
 
-  performGradCheck(IP, result, A, Ex, &inputs, &outputs, 0.001, 0.004);
+  performGradCheck(EE_, result, A, Ex, &inputs, &outputs, 0.001, 0.004);
 }
 
-TEST(Network, gradientCheckConv) { gradientCheckGroupConv(1, 1); }
+TEST_P(GradCheck, gradientCheckConv) { gradientCheckGroupConv(1, 1, EE_); }
```
Comment (rdzhabarov) [inline tests/unittests/gradCheckTest.cpp]: are not all those groupConv tests? (names of the tests seems odd, is it really gradientCheckDepthwiseConv)

File: lib/Backends/CPU/libjit/libjit_conv.cpp (hunk 4 above)
Comment (beicy) [reply lib/Backends/CPU/libjit/libjit_conv.cpp]: Actually, it is the format after running format.sh

Comment (rdzhabarov) [reply lib/Backends/CPU/libjit/libjit_conv.cpp]: :) haha.

File: tests/unittests/gradCheckTest.cpp (hunk 5 above)
Comment (beicy) [reply tests/unittests/gradCheckTest.cpp]: The test are updated by https://github.com/pytorch/glow/commit/361dc5a156496f1d25083cec70f82335cdf6c673#diff-1c1a40d1f65512890dcbc5053e5dd5a3. 
I think the first one is just keep the original test (non-group). The second one is called DepthwiseConv, for depethwise conv, all I know so far is that the number of group should be equal to the input channel.  I need to do more research on this part.

**Comment by nadavrot** [comment]
Nice!   +1 on @artemrakhov 's suggestion to benchmark the perf.
